pkg/utils/yaml_utils/yaml_utils.go

```diff
@@ -100,6 +100,19 @@ func lookupKey(node *yaml.Node, key string) (*yaml.Node, *yaml.Node) {
 	return nil, nil
 }
 
+// Returns the key and value if they were present
+func RemoveKey(node *yaml.Node, key string) (*yaml.Node, *yaml.Node) {
```

Comment:
I think it would make sense to add a test for this, we have tests for most other things in yaml_utils.

Reply:
@ChrisMcD1 I had a need for this function in a PR of mine (#4525), so I stole this function from your PR and added tests for it. (Would also appreciate a review of the rest of the branch if you feel like it.)